### PR TITLE
Make generate_export_header.bzl work for Windows.

### LINKED ISSUE #1351: [BUG] commit 9e47d07 introduced breakage with Bazel under Windows?
Forked from https://github.com/google/benchmark/issues/1345#issuecomment-1041156458:

> ```
> …
> ERROR: C:/users/runneradmin/_bazel_runneradmin/w7nhswsh/external/com_github_google_benchmark/BUILD.bazel:38:11: Compiling src/sleep.cc failed: (Exit 2): cl.exe failed: error executing command C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.30.30705\bin\HostX64\x64\cl.exe /nologo /DCOMPILER_MSVC /DNOMINMAX /D_WIN32_WINNT=0x0601 /D_CRT_SECURE_NO_DEPRECATE ... (remaining 22 arguments skipped)
> C:\users\runneradmin\_bazel_runneradmin\w7nhswsh\execroot\com_googlesource_code_re2\bazel-out\x64_windows-dbg\bin\external\com_github_google_benchmark\_virtual_includes\benchmark\benchmark/benchmark.h(294): error C3861: 'visibility': identifier not found
> C:\users\runneradmin\_bazel_runneradmin\w7nhswsh\execroot\com_googlesource_code_re2\bazel-out\x64_windows-dbg\bin\external\com_github_google_benchmark\_virtual_includes\benchmark\benchmark/benchmark.h(294): error C4430: missing type specifier - int assumed. Note: C++ does not support default-int
> C:\users\runneradmin\_bazel_runneradmin\w7nhswsh\execroot\com_googlesource_code_re2\bazel-out\x64_windows-dbg\bin\external\com_github_google_benchmark\_virtual_includes\benchmark\benchmark/benchmark.h(294): error C2062: type 'void' unexpected
> …
> ```
> 
> (The full logs are [here](https://github.com/google/re2/runs/5211806834?check_suite_focus=true).)

Ping, @sergiud.

### PR DESCRIPTION
**Pull request by junyer**
While I'm here, bring the generated code slightly closer to what CMake
would generate nowadays.

Fixes #1351.

### CODE REVIEW

**Comment by dominichamon** [comment]
LG, waiting for bots.

i don't think we have a bot that would cover this breakage which is a shame, but i'm happy to merge based on the discussions in the issue.

**Comment by dominichamon** [comment]
the failing test is flaky on that bot so i'm going to ignore that :(

**Comment by junyer** [comment]
> i don't think we have a bot that would cover this breakage which is a shame,

Ahh. @sergiud mentioned that in https://github.com/google/benchmark/issues/1351#issuecomment-1044518299. I'm not sure whether [`bazel.yml`](https://github.com/google/benchmark/blob/main/.github/workflows/bazel.yml) just needs a matrix with `ubuntu-latest` and `windows-latest` or what. The caching appears to allow for multiple operating systems already, which is good.

File: BUILD.bazel [hunk 1]
```diff
@@ -55,6 +55,10 @@ cc_library(
         ":windows": [],
         "//conditions:default": posix_copts,
     }),
+    local_defines = select({
+        ":windows": ["BENCHMARK_EXPORTS"],
```
[the remark this answers is not in the document]
Comment (sergiud) [reply BUILD.bazel]: I see Bazel uses a different logic here und uppercases the lib name. I'm not sure if this is a problem.

File: config/generate_export_header.bzl [hunk 2]
```diff
@@ -121,6 +143,8 @@ def generate_export_header(
         name = "__%s_export_h" % lib
     if out == None:
         out = "%s_export.h" % lib
+    if export_import_condition == None:
+        export_import_condition = "%s_EXPORTS" % lib.upper()
```
Comment (sergiud) [inline config/generate_export_header.bzl:147]: Just a minor remark: CMake does not uppercase the `<lib>_EXPORTS` define.

Comment (junyer) [reply config/generate_export_header.bzl]: Fixed. Thanks again!

**Comment by junyer** [comment]
While I'm here, `EXPORT_H` is the generated guard whereas CMake does this:

```
    set(INCLUDE_GUARD_NAME "${EXPORT_MACRO_NAME}_H")
```

That seems much better, no?

**Comment by sergiud** [comment]
Yes, definitely! `EXPORT_H` is too common.

**Comment by junyer** [comment]
Thanks, fixed that as well.

**Comment by dominichamon** [comment]
> > i don't think we have a bot that would cover this breakage which is a shame,
> 
> Ahh. @sergiud mentioned that in [#1351 (comment)](https://github.com/google/benchmark/issues/1351#issuecomment-1044518299). I'm not sure whether [`bazel.yml`](https://github.com/google/benchmark/blob/main/.github/workflows/bazel.yml) just needs a matrix with `ubuntu-latest` and `windows-latest` or what. The caching appears to allow for multiple operating systems already, which is good.

yes, i think it might be enough to add windows to "runs-on".

**Comment by dominichamon** [comment]
@sergiud thanks for all the comments. let me know if you think it's ready to merge :)

**Comment by dominichamon** [comment]
thanks, @junyer !

**Pull request merged**